### ISSUE TITLE
No error is being reported when the closing tag doesn't match the current start tag

### ISSUE THREAD

**Issue by cbettinger** [xmldom/xmldom#45]
I am wondering how I can detect errorneous input when using DOMParse.parseFromString(). To me it seems, that a errorneous element results in just ignoring it. 

Here is a simple demonstration:

**_errorneous.xml (missing '<' at the beginning of line 13):_**

```xml
<?xml version="1.0"?>
<catalog>
	<book id="bk101">
		<author>Gambardella, Matthew</author>
		<title>XML Developer's Guide</title>
		<genre>Computer</genre>
		<price>44.95</price>
		<publish_date>2000-10-01</publish_date>
		<description>An in-depth look at creating applications with XML.</description>
	</book>
	<book id="bk102">
		<author>Ralls, Kim</author>
		title>Midnight Rain</title>
		<genre>Fantasy</genre>
		<price>5.95</price>
		<publish_date>2000-12-16</publish_date>
		<description>A former architect battles corporate zombies, an evil sorceress, and her own childhood to become queen of the world.</description>
	</book>
</catalog>
```

**_parse.js:_**
```js
const fs = require("fs");
const DOMParser = require("xmldom").DOMParser;

let fileContent = fs.readFileSync("errorneous.xml", "utf-8");
let xmlDocument = new DOMParser().parseFromString(fileContent, "text/xml");

console.log(xmlDocument.getElementsByTagName("book").length);   // 2
console.log(xmlDocument.getElementsByTagName("title").length);  // 1
```

**Comment by karfau** [comment]
The current behavior is that it creates the following DOM:
```xml
<?xml version="1.0"?>
<catalog>
        <book id="bk101">
                <author>Gambardella, Matthew</author>
                <title>XML Developer's Guide</title>
                <genre>Computer</genre>
                <price>44.95</price>
                <publish_date>2000-10-01</publish_date>
                <description>An in-depth look at creating applications with XML.</description>
        </book>
        <book id="bk102">
                <author>Ralls, Kim</author>
                title>Midnight Rain
                <genre>Fantasy</genre>
                <price>5.95</price>
                <publish_date>2000-12-16</publish_date>
                <description>A former architect battles corporate zombies, an evil sorceress, and her own childhood to become queen of the world.</description>
        </book>
</catalog>
```
So there is a text node between the second `author` and `genre` tag that is missing the closing tag.
There also isn't any error or warning reported.


- Is the original input well-formed or even valid XML? (I would guess neither.)

- How do browsers handle such a case?
`error on line 13 at column 36: Opening and ending tag mismatch: book line 0 and title`

So potentially this should at least report an error if not throw, which would of course be a breaking change.

**Comment by karfau** [comment]
Still reproducible even on the `next` dist-tag. I think it's worth considering this for 0.9.0.
https://stackblitz.com/edit/js-xmldom-45?devToolsHeight=33&file=index.js

**Comment by karfau** [comment]
The fix was published as version 0.9.0-beta.11 (as the `next` tag).

Please let me know if all your cases are covered.
